ext/opcache/jit/zend_jit_ir.c

```diff
@@ -9061,6 +9067,208 @@ static int zend_jit_init_method_call(zend_jit_ctx         *jit,
 	return 1;
 }
 
+static int zend_jit_init_static_method_call(zend_jit_ctx         *jit,
+                                            const zend_op        *opline,
+                                            uint32_t              b,
+                                            const zend_op_array  *op_array,
+                                            zend_ssa             *ssa,
+                                            const zend_ssa_op    *ssa_op,
+                                            int                   call_level,
+                                            zend_jit_trace_rec   *trace,
+                                            int                   checked_stack)
+{
+	zend_func_info *info = ZEND_FUNC_INFO(op_array);
+	zend_call_info *call_info = NULL;
+	zend_class_entry *ce = NULL;
+	zend_function *func = NULL;
+	ir_ref func_ref, func_ref2, scope_ref, scope_ref2, if_cached, cold_path, ref;
+	ir_ref if_static = IR_UNUSED;
+
+	if (info) {
+		call_info = info->callee_info;
+		while (call_info && call_info->caller_init_opline != opline) {
+			call_info = call_info->next_callee;
+		}
+		if (call_info && call_info->callee_func && !call_info->is_prototype) {
+			func = call_info->callee_func;
+		}
+	}
+
+	if (opline->op1_type == IS_CONST) {
```

Comment:
The "then" and "else" branch are almost identically the same to what happens in `zend_jit_fetch_static_prop`, except for the op number. Perhaps this can be factored out to a separate method?

Reply:
Thanks! Done!